src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -891,6 +893,105 @@ private unsafe void ReadItemsFromSource(ref UnmanagedWriteBuffer writeBuffer, in
             }
         }
 
+        private unsafe class IncomingReplicationAllocator : IDisposable
+        {
+            private readonly DocumentsOperationContext _context;
+            private readonly long _maxSizeForContextUseInBytes;
+            private readonly long _minSizeToAllocateNonContextUseInBytes;
+            public long TotalDocumentsSizeInBytes { get; private set; }
+
+            private List<Allocation> _nativeAllocationList;
+            private Allocation _currentAllocation;
+
+            public IncomingReplicationAllocator(DocumentsOperationContext context, DocumentDatabase database)
+            {
+                _context = context;
+
+                var maxSizeForContextUse = database.Configuration.Replication.MaxSizeToSend * 2 ??
+                              new Size(128, SizeUnit.Megabytes);
+
+                _maxSizeForContextUseInBytes = maxSizeForContextUse.GetValue(SizeUnit.Bytes);
+                var minSizeToNonContextAllocationInMb = PlatformDetails.Is32Bits ? 4 : 16;
+                _minSizeToAllocateNonContextUseInBytes = new Size(minSizeToNonContextAllocationInMb, SizeUnit.Megabytes).GetValue(SizeUnit.Bytes);
+            }
+
+            public byte* AllocateMemoryForDocument(int size)
+            {
+                TotalDocumentsSizeInBytes += size;
+                if (TotalDocumentsSizeInBytes < _maxSizeForContextUseInBytes)
+                {
+                    _context.Allocator.Allocate(size, out var output);
+                    return output.Ptr;
+                }
+
+                if (_currentAllocation == null || _currentAllocation.Free < size)
```

Comment:
`<=` ?

Reply:
Here we don't have enough space on the currently allocated chunk (added a comment)